internal/controller/pkg/revision/deployment.go

```diff
@@ -99,6 +98,16 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 		if cc.Spec.Metadata != nil {
 			d.Spec.Template.Annotations = cc.Spec.Metadata.Annotations
 		}
+		templateLabels := make(map[string]string)
+		if cc.Spec.Metadata != nil {
+			for k, v := range cc.Spec.Metadata.Labels {
+				templateLabels[k] = v
+			}
+		}
+		for k, v := range d.Spec.Selector.MatchLabels { // ensure the template matches the selector
```

Comment:
How would the template matches the selector if no controller config provided?

Reply:
Thanks! Fixed